### ISSUE TITLE
Support `objectOf`

### ISSUE THREAD

**Issue by pellebjerkestrand** [Creuna-Oslo/prop-types-csharp#65]
The `prop-types` [documentation](https://github.com/facebook/prop-types/blob/master/README.md) states:

```
// An object with property values of a certain type
optionalObjectOf: PropTypes.objectOf(PropTypes.number)
```

It would be nice to support this by creating `Dictionary<string, T>` where `T` is whatever we pass to `objectOf`. `string` because keys of JavaScript objects are `string` by default and might also be the only thing that makes sense.

We can get around this for now by using an array and reducing it to an object at runtime, but it seems like unnecessary work for a component to do.

**Comment by asbjornh** [comment]
What’s the use case for this? To me, supporting generics seems to miss the point of this tool, which is making a type safe-ish interface between languages. In this regard generics don’t add much else than complexity as far as I can tell.

**Comment by pellebjerkestrand** [comment]
I might have explained it poorly. There are no generics here.

The use case is having the server produce objects that have keys to be used for lookups. It might, for example, be beneficial to use a `Dictionary<string, Person>` so we can get an object `{ asbjornh: { name: '', email: '' } }`.

**Comment by asbjornh** [comment]
That makes sense. I guess I got caught up in that `T`...

If you can provide an example C# class that uses `Dictionary` that’s is confirmed to build (there might be `using`s needed or something) I can get started on this.

**Comment by pellebjerkestrand** [comment]
Good time to test [Visual Studio for Mac 8](https://visualstudio.microsoft.com/downloads/), I guess.

**Comment by pellebjerkestrand** [comment]
Like so:
<img width="525" alt="cs_dictionary" src="https://user-images.githubusercontent.com/2156573/55461578-0b831800-55f5-11e9-8cde-e8d7779b2841.png">

**Comment by asbjornh** [comment]
Fixed in `1.4.0`
